### DISCUSSION TITLE
什麼是網路爬蟲

### DISCUSSION THREAD

**Discussion by ru0372572** [1101-GS4719-C/Python-Community-GS4719-C#810] [Ideas]
[文章連結-認識網路爬蟲：爬蟲的應用與原理](https://www.webscrapingpro.tw/what-is-web-scraping/#)

「網路爬蟲」是一個透過程式「自動抓取」網站資料的過程，透過爬蟲在網際網路中收集優質的網站與內容，將能更有效率的獲取資訊。

網路爬蟲的原理：
第1步 – 請求網頁內容（爬蟲會請求目標網址，該網頁的伺服器回應網頁的原始碼，通常是HTML文件）
第2步 – 抓取所需資料（爬蟲將 HTML 文件做「解析」並「取出」所需的資料）
![image](https://user-images.githubusercontent.com/113499477/197399599-ea49f43f-3927-4062-89ba-52b2ca85e28c.png)

第3步 – 儲存資料（將取出的資料儲存在 CSV 檔案、Excel 表或是資料庫當中）
![image](https://user-images.githubusercontent.com/113499477/197399573-8aae8adc-a2f3-4254-b6d0-63a0e91dd4a8.png)

然而，過度使用網路爬蟲可能造成別人服務器很大的負擔，所以要當一個「有禮貌」的爬蟲。

1. 遵守 robots.txt 的規範（此文件通常會在網站的根目錄中（www.example.com/robots.txt），它會說明爬蟲應該或是不應該爬的網站內容）
![image](https://user-images.githubusercontent.com/113499477/197399482-086b77b5-6d8c-4ab0-b652-bd19fa57ec13.png)

2. 不造成網站伺服器的負擔（需要在多次的請求之間加入適當的等待延遲）
3. 加入聯繫方式（在 User-agent 標頭中加入公司名稱或電子信箱等資訊）
4. 確認網站是否有提供 API（讓第三方直接取得資料，就不用再去研究網站的 HTML 結構了）

**Comment by Chiway1229** [comment]
所以網路爬蟲也能當成一種資料蒐集的便捷方法嗎
感謝分享這個資訊!

**Comment by ching020721** [comment]
這個網址不僅簡介網路爬蟲，還介紹了使用python進行網路爬蟲的好處，跟課程有所關聯，謝謝分享~

**Comment by HuangHsinTzu** [comment]
感謝分享!
一直聽到爬蟲但都沒有確實了解
爬蟲在大數據的時代裡真的是一項很實用的技術，
在各個領域都能有很棒的應用

**Comment by Kotori0501** [comment]
感謝分享
雖然我對網路爬蟲的資訊並不了解，也沒有實際去做過。
但這網址非常詳細的介紹以及如何正確的使用，對初學者來說是不錯的。
看完之後可以更加了解自己所處的科系又可以多一種方式去實際應用與實踐了。

**Comment by evanchou1204** [comment]
身為python新手，看完這篇對網路爬蟲有了全新的認識，還多知道了一個套件可以用來製作爬蟲，感謝分享！

**Comment by Doreen331** [comment]
之前一直聽到爬蟲這個名詞，對他也只有粗淺的認識，只知道爬蟲是可以去抓網路上的資料的工具，看完這篇之後又學到了更多，之後可以將爬蟲應用到更多地方，不論是作業或是之後的專題

**Comment by dabbie109303501** [comment]
學python就是想學爬蟲，感續分享學習管道和相關套件

**Comment by a45320614** [comment]
感謝分享Python爬蟲的基礎原理，對於剛接觸Python的我幫助很大!

**Comment by Zoe06060606** [comment]
之前常常聽到爬蟲這個詞，現在終於知道他的意義了

**Comment by WEI-CHEN-CHUN** [comment]
很棒的介紹，也提到了如何正確適當的使用他人的網站。

**Comment by Zoe06060606** [comment]
之前一直都有聽過這一項技術 但一直以來都不知道其中的作用原理 這篇文章讓我了解了不少

**Comment by StevenWanglolz** [comment]
上完老師教的網路爬蟲有點跟不上，剛好在這邊複習一下~

**Comment by poiu6fi** [comment]
現在很多網站都會擋爬蟲，導致很多想抓的東西都很難抓下來

**Comment by 710388** [comment]
我也不知道但是在上網家凱老師的課以後感覺有大致的觀念了

**Comment by NcuJerryKao** [comment]
了解網路爬蟲的原理和應用後，這裡有一些建議回覆：

尊重網站所有者的權益：使用網路爬蟲時，要確保不會對目標網站造成過大的負擔，避免影響其正常運作。

遵守網站的規範：爬蟲應該遵守 robots.txt 檔案中所設定的規則，避免爬取不應該取得的資訊，以尊重網站所有者的意願。

控制請求頻率：在爬取資料時，應該控制請求的頻率，避免短時間內大量的請求對伺服器造成負擔，這可以通過加入適當的等待延遲來實現。

加入聯繫方式：在爬蟲的 User-agent 標頭中加入相關的聯繫方式，如公司名稱或電子信箱，以便網站所有者可以聯繫到你，如果有任何問題或疑慮。

考慮使用 API：在可能的情況下，優先考慮使用網站提供的 API 來獲取資料，這是一種更加正式和合法的方式，也可以避免對網站的不必要負擔。

這些建議將有助於你成為一個負責任的爬蟲使用者，同時尊重並遵守網站所有者的權益和規範。
